client/src/components/GDateTime.vue

```diff
@@ -0,0 +1,71 @@
+<script setup lang="ts">
+import { BFormInput, BInputGroup } from "bootstrap-vue";
+import { Tuple } from "types/utilityTypes";
+import { computed } from "vue";
+
+const props = defineProps<{
+    value: Date;
+}>();
+
+const emit = defineEmits<{
+    (e: "input", value: Date): void;
+}>();
+
+const date = computed(() => {
+    const year = props.value.getFullYear().toString().padStart(4, "0");
+    const month = (props.value.getMonth() + 1).toString().padStart(2, "0");
+    const day = props.value.getDate().toString().padStart(2, "0");
+
+    return `${year}-${month}-${day}`;
+});
+
+const time = computed(() => {
+    const hours = props.value.getHours().toString().padStart(2, "0");
+    const minutes = props.value.getMinutes().toString().padStart(2, "0");
+
+    return `${hours}:${minutes}`;
+});
+
+function updateDate(newDate: string) {
+    const matches = newDate.match(/(\d{4})-(\d{2})-(\d{2})/);
+
+    if (matches?.length && matches.length >= 4) {
+        const [_v, year, month, day] = matches as Tuple<4, string>;
+
+        const date = new Date(props.value);
+
+        try {
+            date.setFullYear(parseInt(year));
+            date.setDate(1);
+            date.setMonth(0);
+            date.setMonth(parseInt(month) - 1);
+            date.setDate(parseInt(day));
+        } finally {
+            emit("input", date);
```

Comment:
This might require a bit of tightening in the validation. It looks like you can produce an invalid date if you try a bit harder. The time input seems fine as it is done in a single `setHours`, but the date can be invalid and it will not correct it or complain:

![image](https://github.com/galaxyproject/galaxy/assets/46503462/cd009243-8ae7-4481-ac43-82eb5cb3b065)

I tested this in the downstream PR #16278

Reply:
Ohh, looking at it more closely it seems it is a "problem" (or expected behavior) of the underlying [BFormInput](https://bootstrap-vue.org/docs/components/form-input#input-type):

![image](https://github.com/galaxyproject/galaxy/assets/46503462/39904c88-15d3-46b9-8655-208df29300f1)

So you can ignore it.